tutorials/distributed-ml/torch-tutorial-GAN/train.py

```diff
@@ -0,0 +1,471 @@
+        fake_images = self.generator(noise)
+        fake_images_grid = torchvision.utils.make_grid(
+            fake_images, normalize=True)
+        fig, ax = plt.subplots(figsize=(8, 8))
+        ax.set_axis_off()
+        ax.set_title(f'Fake images for epoch {epoch}')
+        ax.imshow(np.transpose(fake_images_grid.cpu().numpy(), (1, 2, 0)))
+        self.log(
+            item=fig,
+            identifier=f'fake_images_epoch_{epoch}.png',
+            kind='figure',
+            step=epoch,
+        )
+
+
+def main():
+    parser = argparse.ArgumentParser(description='PyTorch MNIST GAN Example')
+    parser.add_argument('--batch-size', type=int, default=128,
+                        help='input batch size for training (default: 128)')
+    parser.add_argument('--epochs', type=int, default=15,
+                        help='number of epochs to train (default: 15)')
+    parser.add_argument('--strategy', type=str, default='ddp',
+                        help='distributed strategy (default=ddp)')
+    parser.add_argument('--lr', type=float, default=0.0002,
+                        help='learning rate (default: 1.0)')
+    parser.add_argument('--seed', type=int, default=1,
+                        help='random seed (default: 1)')
+    parser.add_argument(
+        '--ckpt-interval', type=int, default=2,
+        help='how many batches to wait before logging training status')
+    args = parser.parse_args()
+
+    torch.manual_seed(args.seed)
+
+    # Dataset
+    transform = transforms.Compose([
+        transforms.Resize(64),
+        transforms.ToTensor(),
+        transforms.Normalize((0.5,), (0.5,))
+    ])
+
+    train_dataset = datasets.MNIST(
+        '../data', train=True, download=True, transform=transform)
+    validation_dataset = datasets.MNIST(
+        '../data', train=False, transform=transform)
+
+    def weights_init(m):
+        classname = m.__class__.__name__
+        if classname.find('Conv') != -1:
+            m.weight.data.normal_(0.0, 0.02)
+        elif classname.find('BatchNorm') != -1:
+            m.weight.data.normal_(1.0, 0.02)
+            m.bias.data.fill_(0)
+
+    # Models
+    netG = Generator().to(torch.device("cuda" if torch.cuda.is_available()
+                                       else "cpu "))
+    netG.apply(weights_init)
+    netD = Discriminator().to(torch.device("cuda" if torch.cuda.is_available()
+                                           else "cpu"))
+    netD.apply(weights_init)
+
+    # Training configuration
+    training_config = TrainingConfiguration(
+        batch_size=args.batch_size,
+        lr=args.lr,
+        epochs=args.epochs,
+        loss='cross_entropy'
+    )
+
+    # Logger
+    logger = MLFlowLogger(experiment_name='Distributed GAN MNIST', log_freq=10)
+
+    metrics = {
+        'accuracy': torchmetrics.Accuracy(task='multiclass', num_classes=10),
+        'precision': torchmetrics.Precision(task='multiclass', num_classes=10)
+    }
```

Comment:
you can actually use these metrics to compute the discrimintator precision, or just remove this dict as it is not used

Reply:
Oh yes its not being used, let me remove it.